staging/src/k8s.io/legacy-cloud-providers/aws/aws_fakes.go

```diff
@@ -333,7 +335,10 @@ func (m *FakeMetadata) GetMetadata(key string) (string, error) {
 		return aws.StringValue(i.PublicIpAddress), nil
 	} else if strings.HasPrefix(key, networkInterfacesPrefix) {
 		if key == networkInterfacesPrefix {
-			return strings.Join(m.aws.networkInterfacesMacs, "/\n") + "/\n", nil
+			macs := make([]string, len(m.aws.networkInterfacesMacs))
+			copy(macs, m.aws.networkInterfacesMacs)
+			sort.Strings(macs)
```

Comment:
Really want a numeric sort here.

Reply:
I should add a comment; this is specifically to emulate the observed behavior of AWS; it returns the interface MACs sorted lexically by MAC address, rather than in device-number order. (And in order to test the fix in `TestNodeAddressesWithMetadata`, we need `FakeMetadata` to return the MACs in non-device-number order.)

Reply:
Understood; my point was that if you have more than 9 or 10 interfaces (which I guess doesn't really matter for testing), the sort needs to be numeric rather than lexical.

Does AWS make any promises about ordering of device numbers -- that that won't change, and in particular, that a lower-numbered device won't be added later?  If so, a link to that doc would be good; if not, perhaps a bug needs to be filed against AWS.

Reply:
No, I mean, the sorting isn't based on the device-number at all. We're just sorting the MAC addresses into alphabetical order by MAC address. Why would you want a list of MAC addresses in alphabetical order? I don't know, but that's apparently what AWS gives you.

> Does AWS make any promises about ordering of device numbers

https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/ec2-instance-metadata.html#instancedata-data-categories documents that `device-number` is

> The unique device number associated with that interface. The device number corresponds to the device name; for example, a `device-number` of 2 is for the eth2 device. This category corresponds to the `DeviceIndex` and `device-index` fields that are used by the Amazon EC2 API and the EC2 commands for the AWS CLI. 

https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/using-eni.html says that

> Every instance in a VPC has a default network interface, called the primary network interface (eth0). You cannot detach a primary network interface from an instance. You can create and attach additional network interfaces.

https://docs.aws.amazon.com/cli/latest/reference/ec2/attach-network-interface.html documents the CLI `attach-network-interface` command, which has a required `--device-index` parameter. So, when attaching a new network interface, you explicitly specify the device-number you want it to have. (So, technically, it's *not* guaranteed that a lower-numbered device won't be added later, since the admin could attach eth2 first and then eth1 after that. But if that happens it's because the admin wanted it to happen.)